### PR TITLE
Bump version to Scala Native 0.3.2 and make all unit tests pass

### PR DESCRIPTION
**Pull request by shadaj**
With the PR at https://github.com/scalatest/scalatest/pull/1185, all unit tests now pass in Scala Native!

### CODE REVIEW

File: shared/src/main/scala/squants/Dimension.scala [hunk 1]
```diff
@@ -72,7 +96,11 @@ trait Dimension[A <: Quantity[A]] {
       case _                             ⇒ Failure(QuantityParseException(s"Unable to parse $name", s))
     }
   }
-  private lazy val QuantityString = ("^([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *(" + units.map { u: UnitOfMeasure[A] ⇒ u.symbol }.reduceLeft(_ + "|" + _) + ")$").r
+
+  private lazy val regexUnits = units
```
Comment (derekmorr) [inline shared/src/main/scala/squants/Dimension.scala]: Why was this alias made?

File: shared/src/main/scala/squants/Dimension.scala [hunk 2]
```diff
@@ -55,8 +55,32 @@ trait Dimension[A <: Quantity[A]] {
    * @param value the source string (ie, "10 kW") or tuple (ie, (10, "kW"))
    * @return Try[A]
    */
-  protected def parse(value: Any) = value match {
-    case s: String              => parseString(s)
+  protected def parse(value: Any): Try[A] = value match {
```
Comment (derekmorr) [inline shared/src/main/scala/squants/Dimension.scala]: Why was this change necessary?

File: shared/src/test/scala/squants/thermal/TemperatureSpec.scala [hunk 3]
```diff
@@ -288,7 +288,12 @@ class TemperatureSpec extends FlatSpec
     c.toString(Kelvin) should be("273.15°K")
     c.toString(Fahrenheit) should be("32.0°F")
     c.toString(Celsius) should be("0.0°C")
-    c.toString(Rankine) should be("491.66999999999996°R")
+
+    if (squants.Platform.name == "native") {
```
Comment (derekmorr) [inline shared/src/test/scala/squants/thermal/TemperatureSpec.scala]: Why are these native-specific changes necessary?

File: shared/src/main/scala/squants/thermal/Temperature.scala [hunk 4]
```diff
@@ -141,14 +141,24 @@ object Temperature extends Dimension[Temperature] with BaseDimension {
...
-    val regex = "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *°? *(f|F|c|C|k|K|r|R)".r
-    s match {
-      case regex(value, unit) => unit match {
-        case "f" | "F" => Success(Fahrenheit(value.toDouble))
-        case "c" | "C" => Success(Celsius(value.toDouble))
-        case "k" | "K" => Success(Kelvin(value.toDouble))
-        case "r" | "R" => Success(Rankine(value.toDouble))
-      }
+    val regex = if (squants.Platform.name == "native") {
+      "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *(__DEGREE__)? *(f|F|c|C|k|K|r|R)".r
```
Comment (derekmorr) [inline shared/src/main/scala/squants/thermal/Temperature.scala]: I'm assuming the native platform doesn't support the unicode degree symbol, which is what prompted this change. I'm wondering if we can factor out just that into a conditional to avoid copy/pasting the regex.

Something like:
```scala
val degSymbol = if (squants.Platform.name == "native") "__DEGREE__" else "°"
val regex = s"([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?) *${degSymbol}? *(f|F|c|C|k|K|r|R)".r
```

File: shared/src/main/scala/squants/package.scala [hunk 5]
```diff
@@ -75,6 +75,33 @@ package object squants {
   type Price[A <: Quantity[A]] = squants.market.Price[A]
 
   /**
+    * Helper function to achieve uniform Double formatting over JVM and JS platforms.
+    * Simple Double.toString will format 1.0 as "1.0" on JVM and as "1" on JS
+    * @param d Double number to be formatted
+    * @return
+    */
+  private[squants] def crossFormat(d: Double): String = {
```
Comment (derekmorr) [inline shared/src/main/scala/squants/package.scala]: This version of `crossFormat` isn't the same as before. Did something change in scala native that required us to change?

File: shared/src/main/scala/squants/Dimension.scala (hunk 2 above)
Comment (shadaj) [inline shared/src/main/scala/squants/Dimension.scala]: Scala Native can't handle regexes with unicode yet, so I added an alternate parsing method for native only.

File: shared/src/main/scala/squants/Dimension.scala (hunk 1 above)
Comment (shadaj) [inline shared/src/main/scala/squants/Dimension.scala]: Oops, that was left behind from figuring out how to handle unicode, I'll switch it back to how it was before.

File: shared/src/main/scala/squants/package.scala (hunk 5 above)
Comment (shadaj) [inline shared/src/main/scala/squants/package.scala]: Scala Native now supports formatting, so that can now be shared across platforms, but with the toString implementation on Double right now Scala Native includes trailing zeroes, which would break a lot of unit tests.

File: shared/src/main/scala/squants/thermal/Temperature.scala (hunk 4 above)
Comment (shadaj) [inline shared/src/main/scala/squants/thermal/Temperature.scala]: That sounds good, I'll make that change.

File: shared/src/test/scala/squants/thermal/TemperatureSpec.scala (hunk 3 above)
Comment (shadaj) [inline shared/src/test/scala/squants/thermal/TemperatureSpec.scala]: With Scala Native, the Double toString ends up rounding to "491.67", so until that behavior is fixed I adjusted the tests.

**Comment by shadaj** [comment]
Looks like we won't need the patches for regexes soon, with https://github.com/scala-native/scala-native/pull/946/files on its way!

**Comment by derekmorr** [comment]
Once the tests pass for this PR, we should bump the Scala Native version to 0.3.3, which was just release today.

**Comment by shadaj** [comment]
@derekmorr I'm going to close this PR and work on upgrading directly to Scala Native 0.3.3, since that fixes many of the issues with regexes
